src/mapping/RadialGeoMultiscaleMapping.hpp

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include <vector>
+#include "logging/Logger.hpp"
+#include "mapping/Mapping.hpp"
+
+namespace precice {
+namespace mapping {
+
+/// Geometric multiscale mapping in axial direction
+class RadialGeoMultiscaleMapping : public Mapping {
+public:
+  /**
+   * @brief Geometric multiscale nature of the mapping (spread or collect).
+   *
+   * A geometric multiscale mapping can either go from the 1D to the 2D/3D solver. Then, we call it "spread".
+   * Or from the 2D/3D to the 1D solver, which we call "collect".
+   */
+  enum MultiscaleType {
+    SPREAD,
+    COLLECT
+  };
+  enum RadialAxis {
+    X,
+    Y,
+    Z
+  };
+
+  /**
+   * @brief Constructor.
+   *
+   * @param[in] constraint Specifies mapping to be consistent or conservative.
+   * @param[in] dimensions Dimensionality of the meshes
+   * @param[in] type Geometric multiscale type of the mapping
+   * @param[in] axis Main axis along which radial geometric multiscale coupling happens
+   */
+  RadialGeoMultiscaleMapping(Constraint constraint, int dimensions, MultiscaleType type, RadialAxis axis);
+
+  /// Destructor, empty.
+  virtual ~RadialGeoMultiscaleMapping() {}
+
+  /// Computes the mapping coefficients from the in- and output mesh.
+  void computeMapping() override;
+
+  /// Returns true, if computeMapping() has been called.
+  // bool hasComputedMapping() const override; // check if needed at all
```

Comment:
Reminder to check

Reply:
Has been commented out for a long time, so not needed. Removed in the next commit.